app/src/main/java/is/xyz/mpv/preferences/PreferenceActivity.kt

```diff
@@ -0,0 +1,167 @@
+package `is`.xyz.mpv.preferences
+
+import android.content.SharedPreferences
+import android.os.Build
+import android.os.Bundle
+import android.view.MenuItem
+import android.widget.FrameLayout
+import androidx.activity.enableEdgeToEdge
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.PreferenceManager
+import androidx.preference.SwitchPreferenceCompat
+import com.google.android.material.color.DynamicColors
+import `is`.xyz.mpv.R
+
+
+class PreferenceActivity : AppCompatActivity(),
+    PreferenceFragmentCompat.OnPreferenceStartFragmentCallback,
+    SharedPreferences.OnSharedPreferenceChangeListener {
+    private lateinit var preferences: SharedPreferences
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        preferences = PreferenceManager.getDefaultSharedPreferences(this)
+        preferences.registerOnSharedPreferenceChangeListener(this)
+        if (preferences.getBoolean("material_you_theming", false)) {
+            DynamicColors.applyToActivityIfAvailable(this)
+        }
+
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+
+        val frameLayout = FrameLayout(this).apply {
+            id = R.id.main
+            layoutParams = FrameLayout.LayoutParams(
+                FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.MATCH_PARENT
+            )
+        }
+        frameLayout.id = R.id.main
+        setContentView(frameLayout)
+        supportActionBar?.elevation = 0F
+        ViewCompat.setOnApplyWindowInsetsListener(frameLayout) { v, insets ->
+            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
+            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
+            insets
+        }
+
+        if (savedInstanceState != null) {
+            title = savedInstanceState.getCharSequence("title")
```

Comment:
does the preference fragment set the activity title?

Reply:
Yep. the title will switch back and forth with the Preference sections. 
this savedInstance part was pulled from previous codebase. title acts weird sometimes without it